### PR TITLE
Implement Blob#loc

### PR DESCRIPTION
**Pull request by vmg**
Pretty straightforward. Allows us to count lines on a blob without getting it into Rubyland.

cc @aroben

### CODE REVIEW

**Comment by arthurschreiber** [comment]
LGTM. :+1:

File: ext/rugged/rugged_blob.c [hunk 1]
```diff
@@ -317,6 +317,44 @@ static VALUE rb_git_blob_from_io(int argc, VALUE *argv, VALUE klass)
...
+	data_end = data + git_blob_rawsize(blob);
+
+	if (data == data_end)
+		return INT2FIX(0);
+
+	for (; data < data_end; ++data) {
+		if (data[0] == '\n') {
+			loc++;
+		}
+		else if (data[0] == '\r') {
```
Comment (arthurschreiber) [inline ext/rugged/rugged_blob.c]: Maybe I'm missing something, but is there any need for this else condition?

Comment (arthurschreiber) [inline ext/rugged/rugged_blob.c]: Ah, crap. Ignore my comment. :dizzy_face:

**Comment by aroben** [comment]
Thanks!

**Pull request merged**